third_party/blink/renderer/bindings/core/v8/local_window_proxy.cc

```diff
@@ -233,24 +232,24 @@ void LocalWindowProxy::Initialize() {
       InitializeRecordReplay(GetIsolate(), GetFrame(), context);
     }
 
-    if (doInit || GetFrame() == gRecordReplayFrame) {
```

Comment:
Note: This did not handle potential cross-domain (and maybe other types of?) navigation where the frame does not get re-used.